integration_tests/meta_eval_nlg_test.py

```diff
@@ -189,3 +182,74 @@ def test_dataset_level_spearmanr_bootstrap(self) -> None:
         ci = unwrap(corr_metric.calc_confidence_interval(stats, 0.05))
         self.assertAlmostEqual(ci[0], 1, 2)
         self.assertAlmostEqual(ci[1], 1, 2)
+
+
+class MetaEvalNLGNewsroomTest(unittest.TestCase):
+    """
+    Test the NLG metric on newsroom dataset and replicate the reported results from
+    the paper: https://arxiv.org/pdf/2106.11520.pdf
+    """
+
+    artifact_path = os.path.join(test_artifacts_path, "newsroom")
+    predictions_rouge1 = os.path.join(artifact_path, "rouge1_f_predictions.json")
+    predictions_bartscore = os.path.join(
+        artifact_path, "bart_score_cnn_ref_hypo_predictions.json"
+    )
+
+    def test_coherence_rouge1_f(self):
+        loader = get_loader_class(TaskType.meta_evaluation_nlg).from_datalab(
+            dataset=DatalabLoaderOption("meval_newsroom", "coherence"),
+            output_data=self.predictions_rouge1,
+            output_source=Source.local_filesystem,
+            output_file_type=FileType.json,
+        )
+        data = loader.load().samples
+
+        metadata = {
+            "task_name": TaskType.meta_evaluation_nlg.value,
+            "dataset_name": "meval_newsroom",
+            "sub_dataset_name": "coherence",
+            "metric_names": ["SpearmanSampleLevelCorr"],
+        }
+        processor = get_processor_class(TaskType.meta_evaluation_nlg)()
+        sys_info = processor.process(metadata, data)
+        overall_score = (
+            sys_info.results.overall["example"]["SpearmanSampleLevelCorr"]
+            .get_value(Score, "score")
+            .value
+        )
+        self.assertGreater(len(sys_info.results.analyses), 0)
+        self.assertAlmostEqual(
+            overall_score,
+            0.0946,
+            places=3,
+        )
+
+    def test_coherence_bartscore(self):
+        loader = get_loader_class(TaskType.meta_evaluation_nlg).from_datalab(
+            dataset=DatalabLoaderOption("meval_newsroom", "coherence"),
+            output_data=self.predictions_bartscore,
+            output_source=Source.local_filesystem,
+            output_file_type=FileType.json,
+        )
+        data = loader.load().samples
+
+        metadata = {
+            "task_name": TaskType.meta_evaluation_nlg.value,
+            "dataset_name": "meval_newsroom",
+            "sub_dataset_name": "coherence",
+            "metric_names": ["SpearmanSampleLevelCorr"],
+        }
+        processor = get_processor_class(TaskType.meta_evaluation_nlg)()
+        sys_info = processor.process(metadata, data)
+        overall_score = (
+            sys_info.results.overall["example"]["SpearmanSampleLevelCorr"]
+            .get_value(Score, "score")
+            .value
+        )
+        self.assertGreater(len(sys_info.results.analyses), 0)
+        self.assertAlmostEqual(
+            overall_score,
+            0.3157,
```

Comment:
Ditto

Reply:
This result didn't be listed in paper, but are stored in their github repo: implementation section: https://github.com/neulab/BARTScore#reproduce

Reply:
Thanks! I still didn't see where the number 0.3157 came from though. The Spearman numbers in the table were in the range 0.49.

Could you add a little bit more detail to the comment about where the number could be found?

Sorry, I don't want to be pedantic, but it'd be nice to keep this information around and if we don't do it know we'll likely forget later.